dlt/common/json/__init__.py

```diff
@@ -49,6 +54,8 @@ def custom_encode(obj: Any) -> str:
         return dataclasses.asdict(obj)  # type: ignore
     elif isinstance(obj, Enum):
         return obj.value  # type: ignore[no-any-return]
+    elif isinstance(obj, PydanticAnyUrl):
```

Comment:
you need to also test wether pydanticanyurl is defined, see above how it is done with the Basemodel.

Reply:
I can, but going by @rudolfix' comment the implementation of this might become a fully custom hook anyway, no?